site/components/src/main/java/uk/nhs/digital/common/components/apicatalogue/ApiCatalogueComponent.java

```diff
@@ -56,11 +62,33 @@ private Session sessionFrom(final HstRequest request) {
             .orElse(Collections.emptySet());
     }
 
-    private List<?> apiCatalogueLinksFrom(final HstRequest request) {
-        return ((ComponentList) request.getRequestContext().getContentBean()).getBlocks();
+    private List<Internallink> apiCatalogueLinksFrom(final HstRequest request, final Boolean isToShowAll) {
+
+        return isToShowAll ? retrieveAllLinks(request) : removeDeprecatedAndRetiredLinks(request);
```

Comment:
It feels that it would read easier if we could avoid this conditional and avoid passing the `request` object around, instead extracting the collection of links once and filter it in stages.

How about moving this logic to `apiCatalogueLinksFilteredBySelectedTags`?

The `apiCatalogueLinksFilteredBySelectedTags` would then look something like this:
```
    private List<Internallink> apiCatalogueLinksFilteredBySelectedTags(final List<Internallink> links, final Set<String> selectedTags, final boolean showAll) {

        final List<Internallink> linksWithoutRetiredOrDeprecated = links.stream()
            .filter(link -> showAll || !linkHasSelectedTaxonomyKeys(link, ImmutableSet.of("deprecated-api","retired-api")));
        
        if (selectedTags.isEmpty()) {
            return linksWithoutRetiredOrDeprecated.collect(toList());
        }

        return internalLinksWithSelectedTaxonomyKeys(linksWithoutRetiredOrDeprecated, selectedTags).collect(toList());
    }
```

The name of the method should then probably change to simply say 'filtered' without qualifying 'by what', e.g. by dropping the `BySelectedTags` bit. Naturally, this needs testing to see if the change does not mess up the logic.

Notice that I also changed the key names assuming that my proposal communicated on Slack will be accepted.